### ISSUE TITLE
[door icons] fix rotation for doors with more than 90° rotation

### ISSUE THREAD

**Issue by dan-oak** [EtiamNullam/Etiam-ONI-Modpack#27]
hi
here, all the doors have access right permission set to right
their rotations as appear in the screenshot from left-to-right: neutral, 90, 180, 270
so, the arrows should be: right, down, left, up
![image](https://user-images.githubusercontent.com/66323038/183238066-ed74dbb3-21fe-44c2-8ed3-6ff62ecb7402.png)
(ignore the navigation debug, it's a bug in my mod that allows to rotate buildings)

**Comment by dan-oak** [comment]
i have tried to fix the bug and succeeded if you need help but as i also found that no mod for rotatable doors implements the access rights permissions correctly (e.g. for 180 they reassign left to right for some reason, and for 270 the access rights don't work at all) then the bugfix here doesn't even make sense unless we also want to follow this weird pattern of breaking rotational symmetry here and hardcoding no icon for 270. /shrug

**Comment by EtiamNullam** [comment]
Hmm, I guess it would be nice to support more rotations for doors but I don't see much use of it unless for mods where door are not symmetrical for some reason.

So you mean there's no mod that fixes the permissions direction even if there is some that adds more rotations?

Supporting mods that implement direction permission in such hacky way is out of question. Allowing more rotation for this mod is out of scope of this mod. And I don't feel it is so necessary to create another mod for that.

It would be nice to make the icon always fit the door so if you have any progress on that make sure to submit a pull request ;)

**Comment by dan-oak** [comment]
yes, i mean there are only 2 that allow to rotate buildings in general and specifically for doors they allow 360 rotation, but by doing so they break their permissions. by debugging a bit, i came to the conclusion that it's simply not enough to set `PermittedRotations = PermittedRotations.R360;` on door building definitions, as there is no proper handling of access permissions in klei's code for those high angles.
- https://steamcommunity.com/sharedfiles/filedetails/?id=1715709940
- https://steamcommunity.com/sharedfiles/filedetails/?id=2555723006
and each of them have a few unique bugs apart from the access permissions. unfortunately, they don't respond, maybe i'll do the mod which fixes all their bugs one day if they abandoned their mods.

allowing icons to be displayed at R180 and R270 is not that hard, actually, and done pretty much the same way you've already done for 90. you just need to flip some signs and add an extra unit of offset. i'll show you in a pr maybe tomorrow

**Comment by EtiamNullam** [comment]
Thank you! I've implemented non-standard rotations.
